Makefile

```diff
@@ -0,0 +1,121 @@
+	@echo ">> install jsonnet deps via jb"
+	$(JSONNET_BUNDLER) install
+
+JSONNET_FMT := jsonnetfmt -n 2 --max-blank-lines 2 --string-style s --comment-style s -i
+
+.PHONY: format
+format: ## Auto-formats jsonnet files.
+	@which jsonnetfmt 2>/dev/null || ( \
+		echo "Cannot find jsonnetfmt command, please install from https://github.com/google/jsonnet/releases. If your C++ does not support GLIBCXX_3.4.20, please use xxx-in-container target like format-in-container." && exit 1 \
+	)
+
+	@echo ">> formatting jsonnet"
+	@find . -type f -not -path './vendor/*' \( -name '*.libsonnet' -o -name '*.jsonnet' \) | xargs -n 1 -- $(JSONNET_FMT)
+
+.PHONY: gen
+gen: ## Generates manifests for all environments from jsonnet.
+gen: deps
+	@ENV="dev" $(MAKE) gen-env
+	@ENV="base" $(MAKE) gen-env
+	@ENV="openshift" $(MAKE) gen-env
+
+.PHONY: gen-env
+gen-env: ## Generates manifests for environment given in ENV=... environment variable from jsonnet.
+gen-env: $(GO_JSONNET) $(GOJSONTOYAML)
+	@echo ">> making sure to start with a clean 'manifests' dir for $(ENV)"
+	@rm -rf environments/$(ENV)/manifests
+	@mkdir environments/$(ENV)/manifests
+	@echo ">> building manifests for 'main*.jsonnet' files in $(ENV)"
+ifeq ($(ENV),openshift)
+	@$(GO_JSONNET) -J vendor environments/openshift/main-observatorium-template.jsonnet | $(GOJSONTOYAML) >environments/openshift/manifests/observatorium-template.yaml
+	@$(GO_JSONNET) -J vendor environments/openshift/main-jaeger-template.jsonnet | $(GOJSONTOYAML) >environments/openshift/manifests/jaeger-template.yaml
+	@$(GO_JSONNET) -J vendor environments/openshift/main-observatorium-api-template.jsonnet | $(GOJSONTOYAML) >environments/openshift/manifests/observatorium-api-template.yaml
+else
+	@$(GO_JSONNET) -J vendor -m environments/$(ENV)/manifests environments/$(ENV)/main.jsonnet | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml' -- {}
+endif
+	@find environments/$(ENV)/manifests -type f ! -name '*.yaml' -delete
+
+# Check https://github.com/coreos/prometheus-operator/blob/master/scripts/jsonnet/Dockerfile for the image.
+JSONNET_CONTAINER_CMD:=docker run --rm \
+		-u="$(shell id -u):$(shell id -g)" \
+		-v "$(shell go env GOCACHE):/.cache/go-build" \
+		-v "$(PWD):/configuration:Z" \
+		-w "/configuration" \
+		-e USER=deadbeef \
+		-e GO111MODULE=on \
+		quay.io/coreos/jsonnet-ci:release-0.35
+
+.PHONY: format-in-container
+format-in-container: ## Auto-formats jsonnet files in docker contaier.
+	$(JSONNET_CONTAINER_CMD) $(MAKE) $(MFLAGS) format
+
+.PHONY: gen-in-container
+gen-in-container: ## Generates manifests for all environments from jsonnet in docker container.
+	@echo ">> Compiling and generating thanos-mixin"
+	$(JSONNET_CONTAINER_CMD) $(MAKE) $(MFLAGS) JSONNET_BUNDLER='/go/bin/jb' deps
+	$(JSONNET_CONTAINER_CMD) $(MAKE) $(MFLAGS) \
+		JSONNET='/go/bin/jsonnet' \
+		JSONNET_BUNDLER='/go/bin/jb' \
+		GOJSONTOYAML='/go/bin/gojsontoyaml' \
+		gen
+
+$(GO_JSONNET):
```

Comment:
So for these tool dependencies, could we use the new module approach rather than using a script? https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module

The biggest advantage is we can track all our dependencies with `go.mod` and `go.sum` files.
We have already established the same approach for API gateway and telemeter, see [Makefile] (https://github.com/observatorium/observatorium/blob/master/Makefile#L121) and [tools.go](https://github.com/observatorium/observatorium/blob/master/tools.go) for observatorium/observatorium.

Reply:
Hmmm. What do you think merging this approach and tackle it in separate PR? Also would be nice to move Thanos for this as well.

It's not that easy here as the configuration is *not* a go module... so ... ;p It's weird (:

Reply:
Yeah, that's soo true. 🤦‍♂ We don't need it for this repo. Yep, let's merge this.
For Thanos, I'm happy to contribute.

Reply:
:heart: Remember! Don't promise ;p